include/boost/math/differentiation/autodiff.hpp

```diff
@@ -567,13 +567,36 @@ template<typename RealType, size_t Order, size_t... Orders>
 using autodiff_fvar = typename detail::nest_fvar<RealType,Order,Orders...>::type;
 
 template<typename RealType, size_t Order, size_t... Orders>
-autodiff_fvar<RealType,Order,Orders...> make_fvar(const RealType& ca)
-{
+autodiff_fvar<RealType,Order,Orders...> make_fvar(const RealType& ca) {
   return autodiff_fvar<RealType,Order,Orders...>(ca, true);
 }
 
 namespace detail {
 
+template <size_t>
+struct zero : std::integral_constant<size_t,0> {};
+
+template<typename RealType, size_t Order, size_t... Is>
```

Comment:
Is Is intended to stand for integer sequence? I think a clearer name would be LeadingZeros

Reply:
If this is meant to be used as a sequence of leading zeros, I would suggest using `mp11::repeat` or `mp11::repeat_c`. I may be able to take a shot at this later today or tomorrow if I know what the intention is.

Reply:
Yes that is the intention (though I would not call them "leading zeroes" - they are just "zeroes".)

> Is `Is` intended to stand for integer sequence? I think a clearer name would be LeadingZeros

`Is` can stand for integer sequence, or `I` can be index, and `Is` be its plural form. In any case, I don't think it is approproate to rename it to LeadingZeroes or Zeroes because at that point they are not zeroes - it is a sequence 0, 1, 2, ..., N-1. It is `zero<Is>::value...` that is a sequence of zeroes.

Reply:
@kedarbhat I had a read an it looks like mp11::repeat/repeat_c gives a sequence of types rather than a sequence of values. Does this sound right? If so I don't think it's applicable here.

@pulver Is it correct to say that within this method the actual values in ```Is``` are no longer relevant, we just care about their count?

Reply:
> Is it correct to say that within this method the actual values in Is are no longer relevant, we just care about their count?

Yes.

Reply:
> @kedarbhat I had a read an it looks like mp11::repeat/repeat_c gives a sequence of types rather than a sequence of values. Does this sound right? If so I don't think it's applicable here.
> 
> @pulver Is it correct to say that within this method the actual values in `Is` are no longer relevant, we just care about their count?

You're correct; my suggestion for using `mp_repeat`/`mp_repeat_c` was meant to create  type list, which could then be used as Matthew already is use the `zero` struct. I misread the code within `make_fvar_for_tuple`.